src/routes/resources/+page.svelte

```diff
@@ -63,6 +72,37 @@
     return intersection;
   }
 
+  // Update the displayedResourceLimit based on the scroll position
+  function updateDisplayLimit() {
+    const { scrollTop, clientHeight, scrollHeight } = document.documentElement;
+    const currentPosition = scrollTop + clientHeight;
+
+    if (currentPosition >= scrollHeight - SCROLL_THRESHOLD) {
+      displayedResourceLimit += DEFAULT_DISPLAY_LIMIT;
+      console.log("updateLimit");
+    }
+  }
+
+  // Event listener for scroll events and update limit
+  function handleScroll() {
+    scrollPosition = document.documentElement.scrollTop;
+    showButton = scrollPosition >= window.innerHeight * 2;
+    if (scrollPosition >= SCROLL_THRESHOLD) updateDisplayLimit();
+  }
+
+  // Hook into component lifecycle events
+  onMount(() => {
+    window.addEventListener("scroll", handleScroll);
+  });
+
+  // for button
+  function scrollToTop() {
+    window.scrollTo({
+      top: 0,
+      behavior: "smooth",
+    });
```

Comment:
Great implementation! Even more succinct than the [w3 schools implementation](https://www.w3schools.com/howto/howto_js_scroll_to_top.asp) I saw :3